hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -773,6 +773,34 @@ onOpenClientRecover headId currentSlot coordinatedHeadState recoverTxId =
  where
   CoordinatedHeadState{pendingDeposits} = coordinatedHeadState
 
+-- | Client request to recover deposited UTxO.
+--
+-- __Transition__: 'ClosedState' → 'ClosedState'
+onClosedClientRecover ::
+  IsChainState tx =>
+  HeadId ->
+  ChainSlot ->
+  ChainStateType tx ->
+  TxIdType tx ->
+  Outcome tx
+onClosedClientRecover headId currentSlot chainState recoverTxId =
+  let maybeDeposited = findBy chainState $ \txid _out -> txid == recoverTxId
+   in case maybeDeposited of
+        Nothing -> Error $ RequireFailed NoMatchingDeposit
+        Just deposited ->
+          causes
+            [ OnChainEffect
+                { postChainTx =
+                    RecoverTx
+                      { headId
+                      , recoverTxId = recoverTxId
+                      , -- XXX: Why is this called deadline?
+                        deadline = currentSlot
```

Comment:
Should: not need to pass in `currentSlot` and not need tracking of it in `Closed` state

If we would have access to the deposit's deadline, we could always just pass in that (as also indicated by the field name here!) to build a valid recover transaction (valid from deadline slot onwards).